### PR TITLE
chart-repo: strip leading and trailing spaces for the repo URL

### PR DESCRIPTION
**Pull request by sameersbn**
Repo sync fails of the specified repo URL is sumbitted with
leading/trailing spaces.

### CODE REVIEW

File: cmd/chart-repo/sync.go [hunk 1]
```diff
@@ -18,10 +18,11 @@ package main
 
 import (
 	"os"
+	"strings"
 
-	"github.com/spf13/cobra"
-	"github.com/sirupsen/logrus"
 	"github.com/kubeapps/common/datastore"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
```
Comment (sameersbn) [inline cmd/chart-repo/sync.go]: this reordering is performed by the GoSublime plugin. Let me know if you want it to be reverted (i prefer that it remains)

**Comment by sameersbn** [comment]
@prydonius i've moved the `TrimSpace` call to `syncRepo`. But I am not sure how I should deal with it in the tests because the spaces from the url need to be trimmed before calling `url.ParseRequestURI` https://github.com/sameersbn/kubeapps/blob/4ada23f8e3b958cfcd4f5196b6a5f2ac844a03d3/cmd/chart-repo/utils_test.go#L156

**Comment by prydonius** [comment]
Ah good point, we could create a function `parseRepoURL` that we can then test?

File: cmd/chart-repo/utils.go [hunk 2]
```diff
@@ -68,18 +74,17 @@ var netClient httpClient = &http.Client{
...
 		return err
 	}
-
 	index, err := fetchRepoIndex(url)
 	if err != nil {
 		return err
 	}
 
-	charts := chartsFromIndex(index, repo{Name: repoName, URL: repoURL})
+	charts := chartsFromIndex(index, repo{Name: repoName, URL: strings.TrimSpace(repoURL)})
```
[the remark this answers is not in the document]
Comment (sameersbn) [reply cmd/chart-repo/utils.go]: good point! fixed!

**Pull request merged**